ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -347,13 +358,33 @@ public void poll(long timeoutMs, List<NetworkSend> sends)
           close(key);
         }
       }
+      completeSslHandshakes();
```

Comment:
Let me try to explain why I have it here. The ssl handshake would complete in either write() read() call for keys returned via nioSelector.selectedKeys(). Which means, we at the end of the while loop (313 - 358), we will have some keys which have completed their handshakes. If we wait until next poll() call which the caller has to make, we add some more additional latency to it and also, we add one more additional poll() call since we haven't added these connections which have completed handshakes to the connected list. Hence the network client still thinks that the connection is not ready yet for the next poll(). For these reasons, I want to have it here instead of above the while loop.